baselines/ippo/ippo_pufferlib.py

```diff
@@ -0,0 +1,218 @@
+    backend = backend_mapping.get(args.vec.backend)
+    if not backend:
+        raise ValueError(
+            "Invalid --vec.backend. Choose from native/serial/multiprocessing/ray."
+        )
+
+    vecenv = pufferlib.vector.make(
+        make_env,
+        num_envs=args.vec.num_envs,
+        num_workers=args.vec.num_workers,
+        batch_size=args.vec.env_batch_size,
+        zero_copy=args.vec.zero_copy,
+        backend=backend,
+    )
+
+    policy = make_policy(vecenv.driver_env).to(args.train.device)
+    args.train.env = args.env
+
+    data = ppo.create(args.train, vecenv, policy, wandb=args.wandb)
+    while data.global_step < args.train.total_timesteps:
+        try:
+            ppo.evaluate(data)  # Rollout
+            ppo.train(data)  # Update policy
+        except KeyboardInterrupt:
+            ppo.close(data)
+            os._exit(0)
+        except Exception:
+            Console().print_exception()
+            os._exit(0)
+
+    ppo.evaluate(data)
+    ppo.close(data)
+
+
+def init_wandb(args, name, id=None, resume=True):
+    """Initialize WandB."""
+    wandb.init(
+        id=id or wandb.util.generate_id(),
+        project=args.wandb_project,
+        entity=args.wandb_entity,
+        group=args.wandb_group,
+        config={"train": dict(args.train), "vec": dict(args.vec)},
+        name=name,
+        save_code=True,
+        resume=resume,
+    )
+    return wandb
+
+
+def sweep(args, project="PPO", sweep_name="my_sweep"):
+    """Initialize a WandB sweep with hyperparameters."""
+    sweep_id = wandb.sweep(
+        sweep=dict(
+            method="random",
+            name=sweep_name,
+            metric={"goal": "maximize", "name": "environment/episode_return"},
+            parameters={
+                "learning_rate": {
+                    "distribution": "log_uniform_values",
+                    "min": 1e-4,
+                    "max": 1e-1,
+                },
+                "batch_size": {"values": [512, 1024, 2048]},
+                "minibatch_size": {"values": [128, 256, 512]},
+            },
+        ),
+        project=project,
+    )
+
+    args.track = True
+    wandb.agent(sweep_id, lambda: train(args), count=100)
+
+
+if __name__ == "__main__":
+    # fmt: off
+    parser = argparse.ArgumentParser(
+        description=f":blowfish: PufferLib PPO [bright_cyan]{pufferlib.__version__}[/] arguments. Shows valid args for your env and policy",
+        formatter_class=RichHelpFormatter,
+        add_help=False,
+    )
+
+    parser.add_argument("--env", type=str, default="gpudrive")
+    parser.add_argument("--data-dir", type=str, default="data/processed/examples")
+    parser.add_argument("--num-worlds", type=int, default=50, help="Number of parallel worlds (environments)")
+    parser.add_argument("--k-unique-scenes", type=int, default=1, help="Number of unique scenes to train on")
+    parser.add_argument("--max-cont-agents", type=int, default=32, help="Number of agents per scenario")
```

Comment:
oh boy we should make this a config at some point

Reply:
Yeah, I can also refactor this to add all of this in a yaml or dataclass and use `typer` or pyrallis to make it work with the command linke?

Reply:
PufferLib has a nice config system with 0 deps. We use .ini files like this:
https://github.com/PufferAI/PufferLib/blob/dev/config/gpudrive.ini (dated, from my original integration)